### PR TITLE
Initial support for TKC GodSpeed75

### PR DESCRIPTION
**Pull request by TerryMathews**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tkc/godspeed75/rules.mk [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+MOUSEKEY_ENABLE = no # Mouse keys
+EXTRAKEY_ENABLE = yes # Audio control and System control
+CONSOLE_ENABLE = no  # Console for debug
+COMMAND_ENABLE = no    # Commands for debug and configuration
+SLEEP_LED_ENABLE = no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes     # USB Nkey Rollover
+AUDIO_ENABLE = no
+RGBLIGHT_ENABLE = no
+SERIAL_LINK_ENABLE = no
+
```
Comment (zvecr) [inline keyboards/tkc/godspeed75/rules.mk:21]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/tkc/godspeed75/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+The Key Company GodSpeed75 is a  is a 75% top mount keyboard featuring a USB-C daughterboard and USB-A hub utilizing the STM32F072 microcontroller.
+
+* Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)
+* Hardware Supported: TKC GodSpeed75
+* Hardware Availability: [TheKey.Company](https://thekey.company/collections/candybar)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make tkc/godspeed75:default:dfu-util
```
Comment (zvecr) [inline keyboards/tkc/godspeed75/readme.md:13]:
```suggestion
    make tkc/godspeed75:default:flash
```

File: keyboards/tkc/godspeed75/bootloader_defs.h [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/* Address for jumping to bootloader on STM32 chips. */
```
Comment (zvecr) [inline keyboards/tkc/godspeed75/bootloader_defs.h:1]: This file can be deleted

File: keyboards/tkc/godspeed75/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define _BL 0
+#define _FL 1
+#define _AL 2
+#define _LL 3
```
Comment (fauxpark) [inline keyboards/tkc/godspeed75/keymaps/via/keymap.c:22]:
```suggestion
enum layer_names {
    _BL,
    _FL,
    _AL,
    _LL
};
```

File: keyboards/tkc/godspeed75/bootloader_defs.h (hunk 3 above)
Comment (TerryMathews) [reply keyboards/tkc/godspeed75/bootloader_defs.h]: Done

File: keyboards/tkc/godspeed75/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+The Key Company GodSpeed75 is a  is a 75% top mount keyboard featuring a USB-C daughterboard and USB-A hub utilizing the STM32F072 microcontroller.
+
+* Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)
+* Hardware Supported: TKC GodSpeed75
+* Hardware Availability: [TheKey.Company](https://thekey.company/collections/candybar)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make tkc/godspeed75:default:flash
```
Comment (fauxpark) [inline keyboards/tkc/godspeed75/readme.md:13]:
```suggestion
    make tkc/godspeed75:default
```

File: keyboards/tkc/godspeed75/config.h [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x544B //TK
+#define PRODUCT_ID      0x0006 //GodSpeed75
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    The Key Company
+#define PRODUCT         GodSpeed75
+#define DESCRIPTION     A premium 75% layout keyboard
```
Comment (fauxpark) [inline keyboards/tkc/godspeed75/config.h:27]:
```suggestion
```

File: keyboards/tkc/godspeed75/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# The Key Company GodSpeed75
+
+![GodSpeed75](https://cdn.shopify.com/s/files/1/1679/2319/products/Godspeed75_white_infinikey-bow_720x.jpg?v=1580230323)
+
+The Key Company GodSpeed75 is a  is a 75% top mount keyboard featuring a USB-C daughterboard and USB-A hub utilizing the STM32F072 microcontroller.
+
+* Keyboard Maintainer: [Terry Mathews](https://github.com/TerryMathews/)
+* Hardware Supported: TKC GodSpeed75
+* Hardware Availability: [TheKey.Company](https://thekey.company/collections/candybar)
```
Comment (noroadsleft) [inline keyboards/tkc/godspeed75/readme.md:9]:
```suggestion
* Hardware Availability: [TheKey.Company](https://thekey.company/products/godspeed75)
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**